### PR TITLE
Expose BoomerAMG options

### PR DESCRIPTION
**Pull request by victorapm**
This PR adds the following BoomerAMG calls to `HyprePreconditioner.cpp` for fine-tuning purposes:

1. `HYPRE_BoomerAMGSetNumPaths`
2. `HYPRE_BoomerAMGSetAggInterpType`
3. `HYPRE_BoomerAMGSetRelaxWt`
4. `HYPRE_BoomerAMGSetPMaxElmts`

In addition, new `enums` are defined for the interpolation types.

TODOs:

- [x] Update meaning of `logLevel` values for the linear solver?

0. No info about solver statistics
1. Linear solver iterations, timings and final residual norm
2. All info from 1 plus preconditioner statistics
3. All info from 1 plus convergence history
4. All info from 1, 2, and 3.

- [x] Add enum for coarsening type?
- [ ] Update documentation
- [ ] Handle the other LAIs

### CODE REVIEW

**Comment by victorapm** [comment]
@castelletto1 Can you take a look at the name changes (interpolation, coarsening...)? Thanks!!

File: src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp [hunk 1]
```diff
@@ -114,91 +114,136 @@ struct LinearSolverParameters
...
+    {
+      default_,           ///< Use LAI's default option
+      extendedIStage2,    ///< Extended+i 2-stage (GPU support)
+      standardStage2,     ///< Standard 2-stage
+      extendedStage2,     ///< Extended 2-stage (GPU support)
+      multipass,          ///< Multipass (GPU support)
+      modifiedExtended,   ///< Modularized Extended (GPU support)
+      modifiedExtendedI,  ///< Modularized Extended+i (GPU support)
+      modifiedExtendedE,  ///< Modularized Extended+e (GPU support)
+      modifiedMultipass   ///< Modularized Multipass (GPU support)
```
Comment (castelletto1) [inline src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp:220]: I personally like spelled-out names. The code becomes very readable.

@rrsettgast @joshua-white what do you think?

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ void hypre::mgr::createMGR( LinearSolverParameters const & params,
   // Hypre's parameters to use MGR as a preconditioner
   GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetTol( precond.ptr, 0.0 ) );
   GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetMaxIter( precond.ptr, 1 ) );
-  GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetPrintLevel( precond.ptr, LvArray::integerConversion< HYPRE_Int >( params.logLevel ) ) );
+  GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetPrintLevel( precond.ptr, LvArray::math::min( LvArray::integerConversion< HYPRE_Int >( params.logLevel - 1), 0 ) ) );
```
Comment (castelletto1) [inline src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp:59]: Just a suggestion to have a consistent style with what you did for instance in `HyprePreconditioner.cpp`.

```suggestion
  HYPRE_Int logLevel = LvArray::math::min( LvArray::integerConversion< HYPRE_Int >( params.logLevel - 1), 0 );
  GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetPrintLevel( precond.ptr, logLevel);
```

**Review by castelletto1 — APPROVED** [review]
Looks good to me, thank you very much!

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp [hunk 3]
```diff
@@ -43,6 +43,10 @@ void HypreInterface::initialize()
 #endif
   HYPRE_SetMemoryLocation( hypre::memoryLocation );
 
+#if defined(HYPRE_USING_MEMORY_TRACKER)
```
[the remark this answers is not in the document]
Comment (victorapm) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp:46]: Off. This was not meant to be here, forgot to delete it. But we can leave it for future memory use tests if you like

[the remark this answers is not in the document]
Comment (rrsettgast) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp:46]: is there overhead?

[the remark this answers is not in the document]
Comment (castelletto1) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp:46]: Maybe let's remove it for now.

File: src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp (hunk 1 above)
Comment (rrsettgast) [reply src/coreComponents/linearAlgebra/utilities/LinearSolverParameters.hpp:220]: I agree

**Review by rrsettgast — APPROVED** [review]

**Comment by castelletto1** [comment]
> - [ ] Update documentation
> - [ ] Handle the other LAIs

I have not seen any update on the documentation. Are you planning to take of it in this PR?

I would postpone handling the other LAIs to another PR.

**Comment by victorapm** [comment]
Not necessarily in this PR but I listed it here since I knew that had to be done... I'm okay with leaving it for another PR if you agree

**Comment by castelletto1** [comment]
I agree, let's merge this PR as is.

File: src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (victorapm) [reply src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp]: Done! We can get back to this later. BTW, that's a lightweight memory usage tracking tool we have in hypre. There's close to zero overhead and it provides some interesting insights...

File: src/coreComponents/linearAlgebra/unitTests/testExternalSolvers.cpp [hunk 4]
```diff
@@ -83,7 +83,7 @@ LinearSolverParameters params_CG_AMG()
   parameters.isSymmetric = true;
   parameters.solverType = LinearSolverParameters::SolverType::cg;
   parameters.preconditionerType = LinearSolverParameters::PreconditionerType::amg;
-  parameters.amg.smootherType = geos::LinearSolverParameters::AMG::SmootherType::fgs;
+  parameters.amg.smootherType = geos::LinearSolverParameters::AMG::SmootherType::sgs;
```
Comment (castelletto1) [inline src/coreComponents/linearAlgebra/unitTests/testExternalSolvers.cpp:86]: @victorapm @klevzoff: do you agree with this change? With `fgs` as a smoother, AMG preconditioned CG does not converge (not surprisingly) in 300 iterations to the required tolerance (1e-8) for the 2D Laplace unit test (problem size 10,000). With a symmetric smoother like `sgs`, CG converges in 6 iterations, as one would expect.

Comment (victorapm) [reply src/coreComponents/linearAlgebra/unitTests/testExternalSolvers.cpp:86]: Yes, that makes much more sense to me

**Pull request merged**